---
title: "Analysis"
author: "Kari"
date: "June 29, 2015"
output: html_document
---


```{r,echo=FALSE,message=FALSE,warning=FALSE}
require(knitr)
require(ggplot2)
require(plyr)
require(reshape2)
require(Biostrings)
opts_chunk$set(fig.align="center",warning=FALSE,echo=FALSE, fig.width=10,fig.height=7)

```

```{r,data_setup,cache=TRUE}
#read in coverage files
reference.fasta<-"../../Brisbane_seq_untranslated.fa"
segments <- fasta.info(reference.fasta)
regions.bed <- data.frame(chr = gsub("[ ].*","", names(segments)), start=1, stop=segments, row.names=NULL)
regions.bed<-mutate(regions.bed,chr=as.character(chr))
regions.bed<-mutate(regions.bed,chr=gsub("N_A","NA",chr))
#notanda

cov1.df<-read.delim("../pipeline_output/061015.cov",stringsAsFactor=F)
cov2.df<-read.delim("../pipeline_output/062915.cov",stringsAsFactor=F)

cov.df<-rbind(cov1.df,cov2.df) #rbind combines rows of multiple files

cov.df$Segment[cov.df$Segment=="N_A"]<-as.character("NA")
cov.df$Segment<-factor(cov.df$Segment,levels=c("PB2","PB1","PA","HA","NP","NA","M","NS"))

#read in csv files

map.min<-30 # range of average read position mapping
map.max<-90
p.cut<-0.01 # P value cut off
freq.cut<-0.005 # Frequency cut off 
summary1.df<-read.csv("../pipeline_output/062915.sum.csv", stringsAsFactors=F) # reads in the data
summary2.df<-read.csv("../pipeline_output/061015.csv", stringsAsFactors=F)
summary.df<-rbind(summary1.df,summary2.df)
summary.df<-subset(summary.df,Read_pos>map.min & Read_pos<map.max & p.val<p.cut & freq.var>freq.cut)
```


#Coverage plots
```{r,coverage, cache=TRUE}

fill.cov<-function(x){
pos<-x$Position
seg<-x$Segment[1]
expected.pos<-seq.int(from=1,to=regions.bed$stop[regions.bed$chr==seg],by=1)
missing<-which(!(expected.pos %in% pos))
rows<-length(missing)
sample<-x$Sample[1]
data.frame(Segment=rep(seg,times=rows),Position=missing,Coverage=rep(0,times=rows),Sample=rep(sample,times=rows))
}
z.cov<-ddply(cov.df,~Sample+Segment,fill.cov)

cov.df<-rbind(cov.df,z.cov)
cov.df<-cov.df[order(cov.df$Sample,cov.df$Segment,cov.df$Position),] # reorder to account for the added data

prior.seg.length<-c()

for(k in 1:length(regions.bed$chr)){ 
  prior.seg.length[k]<-sum(regions.bed$stop[1:k])  # the end positions of each segment relative to one sequence not including the trimming step
}

prior.seg.length<-c(0,prior.seg.length)


slideFunct <- function(data, window, step){ #dapted from http://coleoguy.blogspot.com/2014/04/sliding-window-analysis.html
  coverage<-data$Coverage
  pos<-data$Position
  # c.pos<-data$concat.pos
  total <- length(coverage)
  spots <- seq(from=1, to=(total-window), by=step)
  result <- data.frame(mean=rep(F,times=length(spots)),pos=rep(F,times=length(spots)))
  for(i in 1:length(spots)){
    result$mean[i] <- mean(coverage[spots[i]:(spots[i]+window)])
    result$pos[i]<-mean(pos[spots[i]:(spots[i]+window)])
    #result$concat.pos[i]<-mean(c.pos[spots[i]:(spots[i]+window)])
  }
  return(result)
}


wind<-200
step<-100
cov.slid.df<-ddply(cov.df,~Sample+Segment,function(x) slideFunct(x,window = wind,step=step))



cov.slid.df<-mutate(cov.slid.df, # the concatenated position relative to the usual order
                    concat.pos=pos+prior.seg.length[match(Segment,regions.bed$chr)]) # position is relative to each segement


cov.plot<-ggplot(cov.slid.df, #!(Sample %in%c("90","91", "93"))
                 mapping=aes(x=as.factor(concat.pos),
                             y=mean,fill=Segment))+geom_boxplot()

cov.plot<-cov.plot+ggtitle("Sequencing coverage")+ylab("Read depth")+xlab("")+xlab("Concatenated Genome Position")
cov.plot<-cov.plot+scale_x_discrete(labels = "")
cov.plot<-cov.plot+theme(axis.title.y = element_text(vjust=1.2))
cov.plot<-cov.plot+theme(legend.position="top",legend.text=element_text(size=18))
cov.plot
cov.plot+scale_y_log10()

pdf("../Figures/cov.pdf")
cov.plot
dev.off()

pdf("../Figures/cov.log.pdf")
cov.plot+scale_y_log10()
dev.off()

```
#Intrahost Diversity
Diversity just at variant positions

```{r,Diversity_setup,cache=TRUE,message=FALSE,echo=FALSE}
mut_table<-dcast(summary.df,mutation~Id,value.var="freq.var")
mut_table[is.na(mut_table)]<-0 # Setting the NA to 0
mut_table<-as.data.frame(sapply(mut_table,as.numeric))
#write.csv(mut_table,file="./mutant.table.csv")
mut_table<-mut_table[,names(mut_table)!="mutation"]

diversity.df<-subset(summary.df,freq.var<.995)
shannon<-function(x){
  coverage<-x$cov.tst.fw +x$cov.tst.bw
  freq.var<-x$freq.var
  freq.con<-1-freq.var
  sn<- -(freq.var*log(freq.var)/coverage+freq.con*log(freq.con)/coverage)
}

bp<-length(unique(diversity.df$mutation))
simpsonsD<-function(x){
  n.var<-x$n.tst.fw+x$n.tst.bw
  coverage<-x$cov.tst.fw+x$cov.tst.bw
  n.con<-coverage-n.var
  d<-(n.var*(n.var-1)+n.con*(n.con-1))/(coverage*(coverage-1))
  recip<-1/d
}

genome.wideD<-function(x){
  recip<-simpsonsD(x)
  mut.num<-dim(x)[1]
  other.num<-bp-mut.num
  D<-(sum(recip)+other.num)/bp
  }
```



```{r,positional_d,cache=TRUE,message=FALSE,echo=FALSE}
genome.df<-ddply(diversity.df,~Id,function(x){
  summarize(x,pos=pos, chr=chr, mutation=mutation,freq.var=freq.var, simpsonsD=simpsonsD(x),shannon=shannon(x))})

genome.df<-mutate(genome.df, # the concatenated position relative to the usual order
                    concat.pos=pos+prior.seg.length[match(chr,regions.bed$chr)]) # position is relative to each segement


genome.df$Group<-"Placebo"

vaccinated<-c(5,9,11,12,13,14,15,16,19,20,21,23,24,26,28,30,31,32,34,36,43,44,45,46,47,51,53,57,58,59,61,62,63,64,71,73,76,77,79,80,81,82,83,87,88,89,91,92,93,94,99,104,105,106,1,17,22,29,38,41,50,52,60,65,66,69,70,74,78,84,90,96,97,100,102,103)

genome.df$Group[as.numeric(genome.df$Id) %in% vaccinated]<-"Vaccinated"


```

#Plots

```{r,diversity_plots,cache=TRUE,message=FALSE,echo=FALSE}
shannon.p<-ggplot(subset(genome.df,!(Id%in%c("90","91","93"))),aes(x=Group,y=shannon))+geom_boxplot()
shannon.p+scale_y_log10() 

simpsonsD.p<-ggplot(subset(genome.df,!(Id%in%c("90","91","93"))),aes(x=Group,y=log(base=2,simpsonsD)))+geom_boxplot()
simpsonsD.p+scale_y_log10() 

```

Same number of positions between groups

```{r,diversity_equal_pos,cache=TRUE,echo=FALSE}

genome.df<-subset(genome.df,!(Id%in% c("90", "91", "93")))

mutants<-unique(genome.df$mutation)

position.interest<-subset(genome.df,mutation %in% mutants,select=c(pos,chr,concat.pos,mutation))

var.pos.df<-ddply(genome.df, ~Id, function(x){
  missing<-which(!(position.interest$mutation %in% x$mutation))
  missing.df<-position.interest[missing,]
  missing.df$chr=x$chr[1]
  missing.df$freq.var=0
  return(missing.df)
})

missing.df<-mutate(missing.df,pos=pos, chr=chr, mutation=mutation,freq.var=freq.var, simpsonsD=1,shannon=0)

missing.df<-mutate(genome.df, # the concatenated position relative to the usual order
                    concat.pos=pos+prior.seg.length[match(chr,regions.bed$chr)]) # position is relative to each segement


missing.df$Group<-"Placebo"

vaccinated<-c(5,9,11,12,13,14,15,16,19,20,21,23,24,26,28,30,31,32,34,36,43,44,45,46,47,51,53,57,58,59,61,62,63,64,71,73,76,77,79,80,81,82,83,87,88,89,91,92,93,94,99,104,105,106,1,17,22,29,38,41,50,52,60,65,66,69,70,74,78,84,90,96,97,100,102,103)

missing.df$Group[as.numeric(missing.df$Id) %in% vaccinated]<-"Vaccinated"

var.pos.df<-rbind(genome.df,missing.df)

#shannon.p<-ggplot(var.pos.df,aes(x=Group,y=shannon))+geom_boxplot()
#shannon.p+scale_y_log10() 

#simpsonsD.p<-ggplot(subset(var.pos.df,!(Id%in%c("90","91","93"))),aes(x=Group,y=log(base=2,simpsonsD)))+geom_boxplot()
#simpsonsD.p+scale_y_log10() 



```

#frequency of variants (rough plot, has kinks)

```{r,freq.var}
freq.plot<-ggplot(var.pos.df,
mapping=aes(x=as.factor(concat.pos),y=freq.var,color=chr))+geom_point()+facet_wrap(~Group)
freq.plot

```