---
title: "Analysis"
author: "Kari"
date: "June 29, 2015"
output: html_document
---


```{r,echo=FALSE,message=FALSE,warning=FALSE}
require(knitr)
require(plyr)
require(ggplot2)
require(reshape2)
require(Biostrings)
opts_chunk$set(fig.align="center",warning=FALSE,echo=FALSE, fig.width=10,fig.height=7)

```

```{r,data_setup,cache=TRUE}
#read in coverage files
reference.fasta<-"./CalH3N2SeqCont.fa"
segments <- fasta.info(reference.fasta)
regions.bed <- data.frame(chr = gsub("[ ].*","", names(segments)), start=1, stop=segments, row.names=NULL)
regions.bed<-mutate(regions.bed,chr=as.character(chr))


cov.df<-read.csv("./Run1412_all.coverage.csv",stringsAsFactor=F)


#read in csv files

map.min<-30 # range of average read position mapping
map.max<-90
p.cut<-0.01 # P value cut off
freq.cut<-0.00 # Frequency cut off 
summary.df<-read.csv("Run1412_all.sum.csv", stringsAsFactors=F) # reads in the data
summary.df<-subset(summary.df,Read_pos>map.min & Read_pos<map.max & p.val<p.cut & freq.var>freq.cut)
```


#Coverage plots
```{r,coverage, cache=TRUE}
slideFunct <- function(data, window, step){ #dapted from http://coleoguy.blogspot.com/2014/04/sliding-window-analysis.html
  coverage<-data$coverage
  pos<-data$concat.pos
  # c.pos<-data$concat.pos
  total <- length(coverage)
  spots <- seq(from=1, to=(total-window), by=step)
  result <- data.frame(mean=rep(F,times=length(spots)),pos=rep(F,times=length(spots)))
  for(i in 1:length(spots)){
    result$mean[i] <- mean(coverage[spots[i]:(spots[i]+window)])
    result$pos[i]<-mean(pos[spots[i]:(spots[i]+window)])
    #result$concat.pos[i]<-mean(c.pos[spots[i]:(spots[i]+window)])
  }
  return(result)
}


wind<-200
step<-100
cov.slid.df<-ddply(cov.df,~Id+chr,function(x) slideFunct(x,window = wind,step=step))


cov.plot<-ggplot(cov.slid.df, #!(Sample %in%c("90","91", "93"))
                 mapping=aes(x=as.factor(pos),
                             y=mean,fill=chr))+geom_boxplot()

cov.plot<-cov.plot+ggtitle("Sequencing coverage")+ylab("Read depth")+xlab("")+xlab("Concatenated Genome Position")
cov.plot<-cov.plot+scale_x_discrete(labels = "")
cov.plot<-cov.plot+theme(axis.title.y = element_text(vjust=1.2))
cov.plot<-cov.plot+theme(legend.position="top",legend.text=element_text(size=18))
cov.plot
cov.plot+scale_y_log10()
ggplot(cov.df,aes(x=concat.pos,y=coverage,color=Id))+geom_line()

# pdf("../Figures/cov.pdf")
# cov.plot
# dev.off()
# 
# pdf("../Figures/cov.log.pdf")
# cov.plot+scale_y_log10()
# dev.off()

```


```{r}
ggplot(subset(cov.df,Id=="CAcontrol"),aes(x=concat.pos,y=coverage,color=Id))+geom_line()

ggplot(summary.df,mapping=aes(x=freq.var))+geom_histogram(color='white',binwidth=0.05)+scale_y_log10()+scale_x_log10()

ggplot(summary.df,mapping=aes(x=pos,y=freq.var))+geom_point()+facet_wrap(~chr)+scale_y_log10()

```
